lib/class-wp-json-comments-controller.php

```diff
@@ -0,0 +1,470 @@
+		}
+
+		if ( ! comments_open( $post->ID ) ) {
+			return new WP_Error( 'json_user_cannot_create', __( 'Sorry, the comments are closed for this post.' ), array( 'status' => 401 ) );
+		}
+
+		$prepared_comment = $this->prepare_item_for_database( $request );
+		$prepared_comment['comment_approved'] = wp_allow_comment( $prepared_comment );
+
+		$prepared_comment = apply_filters( 'json_pre_insert_comment', $prepared_comment, $request );
+
+		$comment_id = wp_insert_comment( $prepared_comment );
+		if ( ! $comment_id ) {
+			return new WP_Error( 'json_comment_failed_create', __( 'Creating comment failed.' ), array( 'status' => 500 ) );
+		}
+
+		$new_comment = get_comment( $comment_id );
+		$response = $this->prepare_item_for_response( $new_comment, array( 'context' => 'edit' ) );
+		$response = json_ensure_response( $response );
+
+		$response->set_status( 201 );
+		$response->header( 'Location', json_url( '/wp/comments/' . $comment_id ) );
+
+		return $response;
+	}
+
+	/**
+	 * Edit a comment
+	 *
+	 * @param  WP_JSON_Request $request Full details about the request.
+	 * @return WP_Error|WP_JSON_Response|mixed
+	 */
+	public function update_item( $request ) {
+		$id = (int) $request['id'];
+
+		$comment = get_comment( $id );
+		if ( empty( $comment ) ) {
+			return new WP_Error( 'json_comment_invalid_id', __( 'Invalid comment ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! $this->check_edit_permission( $comment ) ) {
+			return new WP_Error( 'json_user_cannot_edit_comment', __( 'Sorry, you are not allowed to update this comment.' ), array( 'status' => 401 ) );
+		}
+
+		$prepared_args = $this->prepare_item_for_update( $request );
+
+		if ( empty( $prepared_args ) && isset( $request['status'] ) ) {
+			// only the comment status is being changed.
+			$change = $this->handle_status_change( $request['status'], $comment );
+			if ( ! $change ) {
+				return new WP_Error( 'json_comment_failed_edit', __( 'Updating comment status failed.' ), array( 'status' => 500 ) );
+			}
+		} else {
+			$prepared_args['comment_ID'] = $id;
+
+			$updated = wp_update_comment( $prepared_args );
+			if ( 0 === $updated ) {
+				return new WP_Error( 'json_comment_failed_edit', __( 'Updating comment failed.' ), array( 'status' => 500 ) );
+			}
+
+			if ( isset( $request['status'] ) ) {
+				$this->handle_status_change( $request['status'], $comment );
+			}
+		}
+
+		$response = $this->get_item( array(
+			'id'      => $id,
+			'context' => 'edit',
+		) );
+		$response = json_ensure_response( $response );
+		$response->set_status( 201 );
+		$response->header( 'Location', json_url( '/wp/comments/' . $comment->comment_ID ) );
+
+		return $response;
+	}
+
+	/**
+	 * Delete a comment.
+	 *
+	 * @param  WP_JSON_Request $request Full details about the request.
+	 * @return WP_Error|array|mixed
+	 */
+	public function delete_item( $request ) {
+		$id = (int) $request['id'];
+		$force = isset( $request['force'] ) ? (bool) $request['force'] : false;
```

Comment:
I still think deleting an item should always delete an item. If you want to trash the item, it should be `PUT status=trash`